src/commands/lambda/__tests__/flare.test.ts

```diff
@@ -15,7 +67,192 @@ describe('lambda flare', () => {
+    expect(output).toMatchSnapshot()
+  })
+
+  it('requests AWS credentials when none are found and logs output correctly', async () => {
+    const cli = makeCli()
+    const context = createMockContext()
+    await cli.run(
+      ['lambda', 'flare', '-f', 'func', '-r', 'us-west-2', '-c', '123', '-e', 'test@test.com'],
+      context as any
+    )
+    const output = context.stdout.toString()
+    expect(output).toMatchSnapshot()
+    expect(requestAWSCredentials).toHaveBeenCalled()
+  })
+
+  it('runs successfully with all required options specified', async () => {
+    const cli = makeCli()
+    const context = createMockContext()
+    const code = await cli.run(
+      ['lambda', 'flare', '-f', 'func', '-r', 'us-west-2', '-c', '123', '-e', 'test@test.com'],
+      context as any
+    )
+    expect(code).toBe(0)
+  })
+
+  it('successfully adds zip file to FormData', async () => {
+    const appendSpy = jest.spyOn(FormData.prototype, 'append')
+    const cli = makeCli()
+    const context = createMockContext()
+    await cli.run(
+      ['lambda', 'flare', '-f', 'func', '-r', 'us-west-2', '-c', '123', '-e', 'test@test.com'],
+      context as any
+    )
+    expect(appendSpy).toHaveBeenCalledWith('flare_file', expect.anything())
+    appendSpy.mockRestore()
+  })
+
+  it('successfully sends request to Datadog', async () => {
+    const postSpy = jest.spyOn(axios, 'post').mockResolvedValue({status: 200})
+    const cli = makeCli()
+    const context = createMockContext()
+    await cli.run(
+      ['lambda', 'flare', '-f', 'func', '-r', 'us-west-2', '-c', '123', '-e', 'test@test.com'],
+      context as any
+    )
+    expect(postSpy).toHaveBeenCalledWith(
+      expect.anything(),
+      expect.anything(),
+      expect.objectContaining({
+        headers: expect.objectContaining({
+          'DD-API-KEY': 'test-api-key',
+        }) as Record<string, string>,
+      })
+    )
+    const output = context.stdout.toString()
+    expect(output).toMatchSnapshot()
+    postSpy.mockRestore()
+  })
+
+  it('does not sends request to Datadog when a dry run', async () => {
+    const postSpy = (axios.post = jest.fn().mockRejectedValue({status: 500}))
+    const cli = makeCli()
+    const context = createMockContext()
+    await cli.run(
+      ['lambda', 'flare', '-f', 'func', '-r', 'us-west-2', '-c', '123', '-e', 'test@test.com', '-d'],
+      context as any
+    )
+    expect(postSpy).not.toHaveBeenCalled()
+    const output = context.stdout.toString()
+    expect(output).toMatchSnapshot()
+    postSpy.mockRestore()
+  })
+
+  it('fails to send request to Datadog', async () => {
+    axios.post = jest.fn().mockRejectedValue({status: 500})
+    const cli = makeCli()
+    const context = createMockContext()
+    await cli.run(
+      ['lambda', 'flare', '-f', 'func', '-r', 'us-west-2', '-c', '123', '-e', 'test@test.com'],
+      context as any
+    )
+    const output = context.stderr.toString()
     expect(output).toMatchSnapshot()
   })
```

Comment:
I've noticed we only test happy paths, but we never actually check for some errors to fail? What would the error message for this look like? Also `fails to send request to Datadog` isn't that descriptive. What about something like `throws an error when post request to Datadog fails`? Also missing the `code` expect to be `1`.